aten/src/ATen/native/cuda/TensorTopK.cu

```diff
@@ -234,9 +234,14 @@ TORCH_IMPL_FUNC(topk_out_cuda)
     topKInfo.sizes[dim] = 1;                                              \
     indicesInfo.sizes[dim] = 1;                                           \
     /* Collapse all other dims */                                         \
-    int collapseInputDim = inputInfo.collapseDims(dim);                   \
-    int collapseTopKDim = topKInfo.collapseDims(dim);                     \
-    int collapseIndicesDim = indicesInfo.collapseDims(dim);               \
+    int collapseInputDim = 0;                                             \
+    int collapseTopKDim = 0;                                              \
+    int collapseIndicesDim = 0;                                           \
+    if (input.dim() > 1) {                                                \
+      int collapseInputDim = inputInfo.collapseDims(dim);                 \
```

Comment:
this doesn't look right, these values are limited to `if` scope.

Reply:
Ah. You are completely right! Thanks.